node/pkg/watchers/solana/client.go

```diff
@@ -341,8 +343,17 @@ func (s *SolanaWatcher) Run(ctx context.Context) error {
 		ContractAddress: contractAddr,
 	})
 
+	// Don't overwrite these fields if they are already set in case there is an old go routine still using them.
```

Comment:
does this actually happen somewhere or is this just a defense?

Reply:
This drew my eye as well. Maybe it's worth logging if `s.ctx` isn't nil? Could be helpful to know if there's a Go routine out there that shouldn't be there.

Reply:
I have not actually seen it happen, it just seems like a theoretical possibility since the watcher spawns go routines to fetch accounts. Maybe those could still be running (and attempting to log or something) when the watcher restarts.

Reply:
These parameters will always be non-null on a watcher restart. It doesn't mean there is a go routine left around. The code previously just blindly overwrote the values on a restart when it didn't need to. The context is the one passed in from node.go, so it is valid for the life of the program, and once the logger is created, there is no reason to recreate it. It *probably* didn't hurt to blindly overwrite them, but it seemed like a good idea not to. 🤷

Reply:
Sounds good -- in that case does it make sense to rephrase the code comment? IMO it would be nice to preserve what you just wrote.